src/comodo/jaxsimSimulator/jaxsimSimulator.py

```diff
@@ -0,0 +1,465 @@
+        visualization_mode: Optional[str] = None,
+    ) -> None:
+        # ==== Initialize simulator model and data ====
+
+        self._dt = dt
+        self._contact_model_type = contact_model_type
+
+        match contact_model_type:
+            case ContactModelEnum.RIGID:
+                contact_model = RigidContacts()
+            case ContactModelEnum.RELAXED_RIGID:
+                contact_model = RelaxedRigidContacts()
+            case ContactModelEnum.VISCO_ELASTIC:
+                contact_model = ViscoElasticContacts()
+            case ContactModelEnum.SOFT:
+                contact_model = SoftContacts()
+            case _:
+                raise ValueError(f"Invalid contact model type: {contact_model_type}")
+
+        model = js.model.JaxSimModel.build_from_model_description(
+            model_description=robot_model.urdf_string,
+            model_name=robot_model.robot_name,
+            contact_model=contact_model,
+        )
+
+        model = js.model.reduce(
+            model=model,
+            considered_joints=robot_model.joint_name_list,
+        )
+
+        self._model = model
+
+        if contact_params is None:
+            match contact_model_type:
+                case ContactModelEnum.RIGID:
+                    contact_params = RigidContactsParams.build(mu=0.5, K=1.0e4, D=1.0e2)
+                case ContactModelEnum.RELAXED_RIGID:
+                    contact_params = RelaxedRigidContactsParams.build(
+                        mu=0.8, time_constant=0.005
+                    )
+                case ContactModelEnum.VISCO_ELASTIC | ContactModelEnum.SOFT:
+                    contact_params = js.contact.estimate_good_soft_contacts_parameters(
+                        model=self._model,
+                        number_of_active_collidable_points_steady_state=16,
+                        max_penetration=0.002,
+                        damping_ratio=1.0,
+                        static_friction_coefficient=1.0,
+                    )
+                case _:
+                    raise ValueError(
+                        f"Invalid contact model type: {contact_model_type}"
+                    )
+
+        s = s or np.zeros(self._model.dofs())
+
+        self._data = js.data.JaxSimModelData.build(
+            model=self._model,
+            velocity_representation=VelRepr.Mixed,
+            base_position=jnp.array(xyz_rpy[:3]),
+            base_quaternion=jnp.array(self._RPY_to_quat(*xyz_rpy[3:])),
+            joint_positions=jnp.array(s),
+            contacts_params=contact_params,
+        )
+
+        if contact_model_type is not ContactModelEnum.VISCO_ELASTIC:
+            self._integrator = integrators.fixed_step.RungeKutta4.build(
+                dynamics=js.ode.wrap_system_dynamics_for_integration(
+                    model=self._model,
+                    data=self._data,
+                    system_dynamics=js.ode.system_dynamics,
+                )
+            )
```

Comment:
Using RK4 is extremely heavy, particularly for `Rigid` and `RelaxedRigid`. Is it necessary? Can't we use either `ForwardEuler` or `Heun2`?

Reply:
Thanks, I'll test how the simulation performs with integrators of smaller order, and if they produce similar results, I will switch to one of them.

Reply:
Done in 2dcea841be59e5a81bc13e8d6bc6485ed322c68a